src/librbd/internal.cc

```diff
@@ -1943,6 +1943,19 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force) {
 	return r;
       }
 
+      if (!old_format) {
+        r = mirror_image_disable_internal(ictx, true);
+        if (r < 0) {
+          lderr(cct) << "error disabling image mirroring: " << cpp_strerror(r)
+                     << dendl;
+          ictx->owner_lock.put_read();
+          ictx->state->close();
+          return r;
+        }
+
+        // TODO: http://tracker.ceph.com/issues/14421
```

Comment:
If I understood correctly, if we remove the image from the cluster that owns the image (primary image), we just execute the code above. Then, it's the rbd-mirror daemon responsibility to check if the image disappears from the mirroring directory and delete the non-primary image.

I'll remove the TODO comment.